element-connector/src/main/java/org/eclipse/californium/elements/CorrelationContext.java

```diff
@@ -17,18 +17,27 @@
  ******************************************************************************/
 package org.eclipse.californium.elements;
 
+import java.util.Set;
+
 /**
- * A container for storing transport specific information about the context
- * in which a message has been sent or received.
+ * A container for storing transport specific information about the context in
+ * which a message has been sent or received.
  */
 public interface CorrelationContext {
 
 	/**
 	 * Gets a value from this context.
 	 * 
 	 * @param key the key to retrieve the value for.
-	 * @return the value or <code>null</code> if this context does not contain
-	 *         a value for the given key.
+	 * @return the value or <code>null</code> if this context does not contain a
+	 *         value for the given key.
 	 */
 	String get(String key);
+
+	/**
+	 * Sends the list of keys available in the context.
+	 *
+	 * @return list of keys set in the context.
```

Comment:
Why are you returning a `Set` if the docs say *Gets a list of ...*?
Is the returned set ordered?